warp10/src/main/java/io/warp10/WarpConfig.java

```diff
@@ -316,13 +327,16 @@ public static Properties readConfig(Reader reader, Properties properties, boolea
 
     br.close();
 
-    if (errorcount > 0) {
-      System.err.println("Aborting due to " + errorcount + " error" + (errorcount > 1 ? "s" : "") + ".");
-      System.exit(-1);
+    if (!linesInError.isEmpty()) {
+      throw new IOException("Malformed lines " + linesInError.toString() + ".");
```

Comment:
Removing the System.exit call changes the behavior for callers other than the main method of WarpConfig, this will most likely break the behavior of many environment which initialize the Warp 10 config through this method, including the integration in Spark/Flink/Pig/KafkaStreams/NiFi and probably others.

Reply:
Changed the methods to take an extra parameter to do a `System.exit` or throw. If not specified, defaults to `System.exit` for backward-compatibility.
This parameter is the first in the methods because of the varargs and consistency between methods.